src/libfetchers/git.cc

```diff
@@ -322,6 +330,22 @@ struct GitInput : Input
         AutoDelete delTmpDir(tmpDir, true);
         PathFilter filter = defaultPathFilter;
 
+        RunOptions checkCommitOpts(
+            "git",
+            { "-C", repoDir, "cat-file", "commit", input->rev->gitRev() }
+        );
+        checkCommitOpts.searchPath = true;
+        checkCommitOpts.mergeStderrToStdout = true;
+
+        auto result = runProgram(checkCommitOpts);
+        if (WEXITSTATUS(result.first) == 128 && result.second.find("bad file") != std::string::npos)
+            throw Error(
+                "Cannot find git-revision in '%s' of repo '%s' on ref '%s'!",
```

Comment:
This should recommend what to do instead.

Reply:
Fixed. It seems as the `fmt` function doesn't do the highlighting of parameters, so I re-added this manually.